backend/src/controllers/v3/authController.ts

```diff
@@ -0,0 +1,275 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+import { Request, Response } from 'express';
+import jwt from 'jsonwebtoken';
+import * as Sentry from '@sentry/node';
+import * as bigintConversion from 'bigint-conversion';
+const jsrp = require('jsrp');
+import { User, LoginSRPDetail } from '../../models';
+import { issueAuthTokens, createToken } from '../../helpers/auth';
+import { checkUserDevice } from '../../helpers/user';
+import { sendMail } from '../../helpers/nodemailer';
+import { TokenService } from '../../services';
+import { EELogService } from '../../ee/services';
+import { BadRequestError, InternalServerError } from '../../utils/errors';
+import {
+    TOKEN_EMAIL_MFA,
+    ACTION_LOGIN
+} from '../../variables';
+import { getChannelFromUserAgent } from '../../utils/posthog'; // TODO: move this
+import {
+    getJwtMfaLifetime,
+    getJwtMfaSecret,
+    getHttpsEnabled,
+    getJwtProviderAuthSecret
+} from '../../config';
+
+declare module 'jsonwebtoken' {
+    export interface ProviderAuthJwtPayload extends jwt.JwtPayload {
+        userId: string;
+    }
+}
+
+/**
+ * Log in user step 1: Return [salt] and [serverPublicKey] as part of step 1 of SRP protocol
+ * @param req
+ * @param res
+ * @returns
+ */
+export const login1 = async (req: Request, res: Response) => {
+    try {
+        const {
+            email,
+            providerAuthToken,
+            clientPublicKey
+        }: {
+            email?: string;
+            clientPublicKey: string,
+            providerAuthToken?: string;
+        } = req.body;
+
+        let userId = '';
+        if (providerAuthToken) {
+            const decodedToken = <jwt.ProviderAuthJwtPayload>(
+                jwt.verify(providerAuthToken, await getJwtProviderAuthSecret())
+            );
+
+            userId = decodedToken.userId;
+        }
+
+        const filter = userId ? {
+            _id: userId,
+        } : {
+            email,
+        }
+
+        const user = await User.findOne(filter).select('+salt +verifier');
+
+        if (!user) throw new Error('Failed to find user');
+
+        const server = new jsrp.server();
+        server.init(
+            {
+                salt: user.salt,
+                verifier: user.verifier
+            },
+            async () => {
+                // generate server-side public key
+                const serverPublicKey = server.getPublicKey();
+                const identifier = userId ? {
+                    userId,
+                } : {
+                    email,
+                }
+
+                await LoginSRPDetail.findOneAndReplace(filter, {
+                    ...identifier,
```

Comment:
Please check this reply

![image](https://user-images.githubusercontent.com/65645666/235606909-432c7cdc-7eaa-4090-8dd1-b5aaa8eb31f7.png)

The reason why I implemented it this way is so that there is a distinction in the SRP login details for those initiated by conventional email or those through a provider.

Reply:
Gotcha. 

I think we should make it so that you can't sign up manually if the same email has been used for Google login previously; we can add into the user settings so that they can enable password-based logins as well in the future.